packages/pluggableWidgets/datagrid-web/src/components/Table.tsx

```diff
@@ -22,30 +22,32 @@ import { useSettings } from "../utils/settings";
 export type TableColumn = Omit<ColumnsType | ColumnsPreviewType, "content" | "attribute">;
 
 export interface TableProps<T> {
+    cellRenderer: (
+        renderWrapper: (children: ReactNode, className?: string) => ReactElement,
+        value: T,
+        columnIndex: number
+    ) => ReactElement;
     className: string;
-    data: T[];
     columns: TableColumn[];
     columnsFilterable: boolean;
     columnsSortable: boolean;
     columnsResizable: boolean;
     columnsDraggable: boolean;
     columnsHidable: boolean;
+    data: T[];
+    emptyPlaceholderRenderer?: (renderWrapper: (children: ReactNode) => ReactElement) => ReactElement;
```

Comment:
Are we doing this fancyness (passing a function that returns React element **and** as a parameter also pass a function that returns React element). Because we want to pass useCallback ?

Reply:
yeap, and it also follows the same as cellRenderer :)